### DISCUSSION TITLE
My new open-source book in draft stage - Istio & Envoy Insider

### DISCUSSION THREAD

**Discussion by labilezhu** [istio/istio#51065] [Show and tell]
![istio-insider-book-cover-en](https://github.com/istio/istio/assets/22267887/a63644a6-e9ff-4a40-bd2b-ee66b121f649)

My new open-source book - Istio and Envoy Insider, has just reached a milestone. So I want to Introduce it.

First of all, I have to say: I'm not an Istio expert, not even an Istio contributor. I wrote this book is "just for fun 😀" . 

## Overview of this book

This book is called Istio & Envoy Insider. It is a book in progress, now in draft stage.

### What this book is about

This book includes: Envoy source code deep dive, in-depth Envoy fundamentals  analysis , Istio fundamentals analysis. But it's not a traditional "deep dive xyz source code" type of book. on the contrary, I have done my best not to directly paste source code in the book. Reading source code is a necessary step to grasp the details of the implementation, but browsing source code in a book is generally a very bad experience. So, this book uses source code navigation diagrams to let readers understand the full picture of the implementation, rather than getting lost in the details of fragmented source code snippets and forgetting the whole picture.

In this book, I've tried to think as systematically as possible from a design and implementation perspective:
- The design and implementation details of Envoy
- Why Istio is what it is
- The Truth Behind Those Magic Configurations: Linux + Envoy
  - How traffic is intercepted to the Envoy using Linux's netfilter technology.
  - How istiod programs the Envoy to fulfill the traffic policies of the Service Mesh.
- What Istio might look like in the future

The book is a collection of thoughts and notes after I've been researching and using Istio for a while. I've just been troubleshooting some Istio/Envoy related functionality and performance issues, and browsing and debugging some Istio/Envoy code.

### Book access address
- [https://istio-insider.mygraphql.com/en](https://istio-insider.mygraphql.com/en)

**Comment by jasonwjdeng** [comment]
Good job. Good book.

**Comment by MincongYM** [comment]
Wonderful book!

**Comment by style95** [comment]
Hi, is there any difference between the kindle edition and the web contents?

**Comment by labilezhu** [comment]
> Hi, is there any difference between the kindle edition and the web contents?

The web edition is keeping updated. So I recommend web edition 😎
